kinto/core/scripts.py

```diff
@@ -88,3 +91,110 @@ def delete_collection(env, bucket_id, collection_id):
+                                                sorting=[OLDEST_FIRST],
+                                                pagination_rules=bucket_pagination,
+                                                limit=BATCH_SIZE)
+        if not buckets:
+            break
+
+        collection_pagination = None
+        for bucket in buckets:
+            bucket_id = bucket['id']
+
+            bucket_record_count = 0
+            bucket_storage_size = record_size(bucket)
+
+            while True:
+                (collections, _) = registry.storage.get_all(
+                    collection_id='collection',
+                    parent_id='/buckets/{}'.format(bucket['id']),
+                    sorting=[OLDEST_FIRST],
+                    pagination_rules=collection_pagination,
+                    limit=BATCH_SIZE)
+
+                if not collections: break
+
+                for collection in collections:
+                    collection_info = rebuild_quotas_collection(env, bucket_id, collection, dry_run)
+                    bucket_record_count += collection_info['record_count']
+                    bucket_storage_size += collection_info['storage_size']
+
+                collection_pagination = [
+                    [Filter('last_modified', collection['last_modified'], COMPARISON.GT)]
+                ]
+
+        # FIXME: Store bucket_record_count, bucket_storage_size
+        logger.info("Bucket {}. Final size: {} records, {} bytes.".format(
+            bucket_id, bucket_record_count, bucket_storage_size))
+
+        bucket_pagination = [
+            [Filter('last_modified', bucket['last_modified'], COMPARISON.GT)]
+        ]
+
+    current_transaction.commit()
+    return 0
+
+
+def rebuild_quotas_collection(env, bucket_id, collection, dry_run=False):
+    """Helper method for rebuild_quotas that updates a single collection."""
+    registry = env['registry']
+    record_pagination = None
+    collection_id = collection['id']
+    collection_record_count = 0
+    collection_storage_size = record_size(collection)
+    while True:
+        (records, _) = registry.storage.get_all(
+            collection_id='record',
+            parent_id='/buckets/{}/collections/{}'.format(bucket_id, collection_id),
+            sorting=[OLDEST_FIRST],
+            pagination_rules=record_pagination,
+            limit=BATCH_SIZE)
+
+        if not records:
+            break
+
+        for record in records:
+            collection_record_count += 1
+            collection_storage_size += record_size(record)
+
+        record_pagination = [
+            [Filter('last_modified', record['last_modified'], COMPARISON.GT)]
+        ]
+
+    logger.info("Bucket {}, collection {}. Final size: {} records, {} bytes.".format(
+        bucket_id, collection_id, collection_record_count, collection_storage_size))
+    return {
+        "record_count": collection_record_count,
+        "storage_size": collection_storage_size,
+    }
```

Comment:
This is the record that will be inserted into the DB so I figured I would just return that. But it's really just because I was being lazy.